src/guided_fuzzing_daemon/storage.py

```diff
@@ -231,11 +232,15 @@ def delete(self, files: Iterable[CloudStorageFile]) -> None:
                     Delete={"Objects": del_keys, "Quiet": True},
                 )
 
-    def iter(self, prefix: PurePosixPath) -> Iterable[CloudStorageFile]:
+    def iter(
+        self, prefix: PurePosixPath, suffix: str | None = None
+    ) -> Iterable[CloudStorageFile]:
         result = self.client.list_objects_v2(
             Bucket=self.bucket_name, Prefix=f"{prefix}/"
         )
         for obj in result.get("Contents", ()):
+            if suffix is not None and not obj["Key"].endswith(suffix):
+                continue
```

Comment:
Not sure what is expected in the fuzzilli buckets.  Should we log these cases?

Reply:
This commit actually isn't even needed. I ended up making suffix only affect queue/corpus uploads and not care about downloads.